### PR TITLE
Refactors the code and implements unit tests for the checklist utility and service

### PR DESCRIPTION
**Pull request by AdiAkhileshSingh15**
## Summary
This PR refactors functions related to the checklist by moving them to the utility module and also adds unit tests for both the checklist service and utility. Minor changes have been made to other parts of the code, including the creation of the project file constant, and replacing `os.homedir` with `os.homedir()` in some places.

**Note**: Unit tests for the `convertImageToBase64` function in the general utils have not been implemented due to the need for generating mock base64 strings for dummy file paths, which are quite LONG. Though, if you have any suggestions to address this, please LMK.
### Reviewer
@lrasmus

### CODE REVIEW

File: test/services/assets/handlers/file.spec.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ describe('services', () => {
         const handler = new FileHandler();
         expect(handler.includeFile('/User/test/Project/.DS_Store')).toBeFalsy();
         expect(handler.includeFile('C:/test/Project/Thumbs.db')).toBeFalsy();
-        expect(handler.includeFile('.statwrap-project.json')).toBeFalsy();
+        expect(handler.includeFile(Constants.StatWrapFiles.PROJECT)).toBeFalsy();
```
Comment (lrasmus) [inline test/services/assets/handlers/file.spec.js:33]: I want it on the record that I really appreciate you fixing *my* mistake of hard-coding constants.  Thank you :)

File: test/services/checklist.spec.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+          throw new Error('Write failed');
+        });
+
+        expect(() => checklistService.writeChecklist(mockProjectPath, checklistData)).toThrow('Write failed');
+      });
+    });
+
+    describe('loadChecklist', () => {
+      it('should return an error if projectPath is null or undefined', (done) => {
+        checklistService.loadChecklist(null, (err, result) => {
```
Comment (lrasmus) [inline test/services/checklist.spec.js:61]: Could we add a similar check for the `undefined` situation too?

File: app/utils/checklist.js [hunk 3]
```diff
@@ -18,4 +20,37 @@ export default class ChecklistUtil {
     });
     return checklist;
   }
+
+  static findAssetLanguagesAndDependencies(asset, languages = {}, dependencies = {}) {
```
Comment (lrasmus) [inline app/utils/checklist.js:24]: Could you provide a brief comment header to describe the shape of `languages` and `dependencies` for callers?

File: test/services/checklist.spec.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+      it('should correctly resolve the project path and write checklist data to a file', () => {
+        fs.writeFileSync = jest.fn();
+
+        checklistService.writeChecklist(mockProjectPath, checklistData);
+
+        expect(fs.writeFileSync).toHaveBeenCalledWith(
+          resolvedProjectPath,
+          JSON.stringify(checklistData)
+        );
```
Comment (lrasmus) [inline test/services/checklist.spec.js:47]: This is failing on Windows, but passing on macOS/Linux because `TEST_PROJECT_PATH` on Windows is not relative to the home directory.

```
    Expected: "C:\\Users\\test\\testProject\\.statwrap\\.statwrap-checklist.json", "[{\"id\":1,\"name\":\"Dependency\",\"statement\":\"All the software dependencies for the project are documented.\",\"answer\":true}]"
    Received: "C:\\testProject\\.statwrap\\.statwrap-checklist.json", "[{\"id\":1,\"name\":\"Dependency\",\"statement\":\"All the software dependencies for the project are documented.\",\"answer\":true}]"

```

File: test/services/checklist.spec.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+          done();
+        });
+      });
+
+      it('should correctly read and parse checklist data from the file', (done) => {
+        fs.existsSync = jest.fn().mockReturnValue(true);
+        fs.readFileSync = jest.fn().mockReturnValue(JSON.stringify(checklistData));
+
+        checklistService.loadChecklist(mockProjectPath, (err, result) => {
+          expect(err).toBeNull();
```
Comment (lrasmus) [inline test/services/checklist.spec.js:83]: I'm not entirely sure why, but this is failing on Windows only, and saying it found the following error:

```
Received: "Error reading or parsing checklist file"
```

File: test/utils/checklist.spec.js [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+      it('should return empty result for non-code assets (data, documentation)', () => {
+        expect(ChecklistUtil.findAssetLanguagesAndDependencies({
+          type: Constants.AssetType.FILE,
+          contentTypes: [Constants.AssetContentType.DATA],
+          uri: 'path/to/file.csv',
+        })).toEqual({
+          languages: [],
+          dependencies: [],
+        });
+      });
```
Comment (lrasmus) [inline test/utils/checklist.spec.js:43]: I like this - I am wondering if we could have another test similar to this one which returns empty results for non-code assets, but we give the asset a file extension that we would expect it to be flagged if it were a file.  E.g., the content type is still DATA, but we name the uri `path/to/file.py`.  This further ensures that both file extension and content type are being used in the decision.

File: test/utils/general.spec.js [hunk 7]
```diff
@@ -1,7 +1,7 @@
 /* eslint-disable prettier/prettier */
 import GeneralUtil from '../../app/utils/general';
 
-describe('services', () => {
+describe('utils', () => {
```
Comment (lrasmus) [inline test/utils/general.spec.js:4]: Thank you for fixing these incorrect labels too (also my fault!!!)

**Review by lrasmus — CHANGES_REQUESTED** [review]
@AdiAkhileshSingh15  - overall this is amazing.  Great coverage of different scenarios in your testing, and again I thank you for improving the underlying StatWrap code base with changes that I should have made myself long ago.   I have just a few suggestions if you could address them before I merge?

File: app/components/ReproChecklist/ChecklistItem/ChecklistItem.js [hunk 8]
```diff
@@ -78,6 +78,8 @@ function ChecklistItem(props) {
     }
 
     setAddAsset(false);
+    setAssetTitle('');
```
[the remark this answers is not in the document]
Comment (AdiAkhileshSingh15) [reply app/components/ReproChecklist/ChecklistItem/ChecklistItem.js:81]: uh oh! i completely overlooked that

File: test/services/assets/handlers/file.spec.js (hunk 1 above)
Comment (AdiAkhileshSingh15) [reply test/services/assets/handlers/file.spec.js:33]: Not at all, it was just a small thing that caught my attention 😆. We all make mistakes!

File: test/services/checklist.spec.js (hunk 2 above)
Comment (AdiAkhileshSingh15) [reply test/services/checklist.spec.js:61]: sure!

File: app/utils/checklist.js (hunk 3 above)
Comment (AdiAkhileshSingh15) [reply app/utils/checklist.js:24]: definitely!

Comment (AdiAkhileshSingh15) [reply app/utils/checklist.js:24]: actually planned to add the comments and make sure the docs for checklist are updated in a later PR. as there are a lot of places where there's a need to explain why it's done a specific way, so I thought let's handle that all with a single code push.

File: test/services/checklist.spec.js (hunk 4 above)
Comment (AdiAkhileshSingh15) [reply test/services/checklist.spec.js:47]: uh oh! that's my mistake! i'll fix that

File: test/services/checklist.spec.js (hunk 5 above)
Comment (AdiAkhileshSingh15) [reply test/services/checklist.spec.js:83]: maybe because of that previous mistake i guess. the project path for windows wasn't set properly

File: test/utils/checklist.spec.js (hunk 6 above)
Comment (AdiAkhileshSingh15) [reply test/utils/checklist.spec.js:43]: ooh, nice one, don't know why i did not do this

Comment (AdiAkhileshSingh15) [reply test/utils/checklist.spec.js:43]: actually, right now, we DO NOT make decisions using both the parameters. only the content type is being used, so for all files having CODE content type will pass through it.
but, i'll add one more check in the function, making it robust.

File: app/utils/checklist.js (hunk 3 above)
Comment (lrasmus) [reply app/utils/checklist.js:24]: Sounds good!

File: app/components/ReproChecklist/ChecklistItem/ChecklistItem.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (lrasmus) [reply app/components/ReproChecklist/ChecklistItem/ChecklistItem.js:81]: No worries, works great now!

File: test/services/checklist.spec.js (hunk 4 above)
Comment (lrasmus) [reply test/services/checklist.spec.js:47]: Works great, thank you!

File: test/services/checklist.spec.js (hunk 5 above)
Comment (lrasmus) [reply test/services/checklist.spec.js:87]: Also working great now, thanks!

**Review by lrasmus — APPROVED** [review]
Looks great!
